scheduler/scheduler.py

```diff
@@ -30,11 +30,17 @@ def checkType(msg):
     elif msg['type'] == 'milestone.reward' or msg['type'] == 'milestone.penalise':
         # Milestone has been rewarded, need to remove from cron
         milestone_id = msg['resource_id']
-        projId = msg['data']['project_id']
+        projId = msg['data']['id']
         milestoneRemove(milestone_id, projId)
-    elif msg['resource_id'] == 'task.add':
+    elif msg['type'] in ["upcoming", "penalise","rollback"]:
```

Comment:
if type is 'penalise' i.e. the milestone 1 day after its due_date and verifier havent verify, you should publish task with event 'overdue'. Currently it publishes a payload of 'rollback' to the projectpolice.
To test: publish this to `scheduler.task.add` queue
```json
{
  "resource_id": "a3cc8e03-07e4-4211-a029-431beed4ef87",
  "type": "penalise",
  "data": {
    "project_id": "a3cc8e03-07e4-4211-a029-431beed4ef87",
    "milestone_id": "92636c2d-d979-4351-b75a-865e3dd3d5b6",
    "payment_id": null
  }
}

payload published to `task_execute`
```json
{
  "resource_id": "a3cc8e03-07e4-4211-a029-431beed4ef87",
  "type": "rollback",
  "data": {
    "project_id": "a3cc8e03-07e4-4211-a029-431beed4ef87",
    "milestone_id": "92636c2d-d979-4351-b75a-865e3dd3d5b6",
    "payment_id": null
  }
}
```

Reply:
![image](https://user-images.githubusercontent.com/18240979/229267101-e1ca8bda-de60-4e7d-a2fc-78565b1c5156.png)
![image](https://user-images.githubusercontent.com/18240979/229267109-b0cca15b-581e-40ad-9751-4cb1564b61a9.png)
![image](https://user-images.githubusercontent.com/18240979/229267111-5ad23001-f342-4ab2-abcd-d00a1133838f.png)

It should already be event = "overdue"

Reply:
Good spot. I have edited the code to reflect this.